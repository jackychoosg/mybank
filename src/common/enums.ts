export enum TransactionAction {
  DEBIT = 'debit',
  CREDIT = 'credit'
}
