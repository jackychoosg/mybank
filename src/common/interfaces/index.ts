import { TransactionAction } from '../enums'

export interface ITransaction {
  id?: number
  accountId: number
  currencyId: number
  amount: number
  timestamp: number
  description: string
  action: TransactionAction
  reference: string
}

export interface IOption {
  id: number
  name: string
}

export interface IAccount {
  id: number
  accountNumber: string
  name: string
}

export interface ICurrency {
  id: number
  code: string
  name: string
}
