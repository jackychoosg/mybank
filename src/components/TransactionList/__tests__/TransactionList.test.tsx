import { render, cleanup } from '@testing-library/react'
import { TransactionAction } from '../../../common/enums'
import { TransactionList } from '../index'
describe('TransactionList component', () => {
  afterEach(cleanup)
  const currencies = [
    {
      id: 1,
      code: 'SGD',
      name: 'Singapore Dollar'
    },
    {
      id: 2,
      code: 'AUD',
      name: 'Australia Dollar'
    }
  ]

  const data = [
    {
      accountId: 1,
      currencyId: 1,
      timestamp: 1625829468,
      reference: '948324085',
      amount: 1000,
      description: 'Testing',
      action: TransactionAction.DEBIT,
      id: 1
    },
    {
      accountId: 2,
      currencyId: 2,
      timestamp: 1625829468,
      reference: '948324085',
      amount: 1000,
      description: 'Testing',
      action: TransactionAction.CREDIT,
      id: 1
    }
  ]
  it('should take a snapshot with Debit data', () => {
    const { asFragment } = render(
      <TransactionList
        currencies={currencies}
        selectedAccount={1}
        rows={data}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should take a snapshot with credit data', () => {
    const { asFragment } = render(
      <TransactionList
        currencies={currencies}
        selectedAccount={2}
        rows={data}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
