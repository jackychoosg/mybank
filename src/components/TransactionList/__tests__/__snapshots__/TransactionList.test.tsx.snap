// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList component should take a snapshot with Debit data 1`] = `
<DocumentFragment>
  <table
    class="sc-bdnxRM frAAFw"
  >
    <thead
      class="sc-gtsrHT koYlQI st-header"
    >
      <tr
        class="sc-eCApnc"
      >
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Transaction Date
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Description
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Action
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Currency
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-jSFjdj st-body"
    >
      <tr
        class="sc-eCApnc"
      >
        <td
          class="sc-hKFxyN koVabA"
        >
          July 9, 2021 7:17 PM
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          Testing
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          debit
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          SGD
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          <label
            class="sc-iCoGMd ksFDXw"
          >
            -$1,000
          </label>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`TransactionList component should take a snapshot with credit data 1`] = `
<DocumentFragment>
  <table
    class="sc-bdnxRM frAAFw"
  >
    <thead
      class="sc-gtsrHT koYlQI st-header"
    >
      <tr
        class="sc-eCApnc"
      >
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Transaction Date
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Description
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Action
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Currency
        </th>
        <th
          class="sc-dlnjwi cvvJUt"
        >
          Amount
        </th>
      </tr>
    </thead>
    <tbody
      class="sc-jSFjdj st-body"
    >
      <tr
        class="sc-eCApnc"
      >
        <td
          class="sc-hKFxyN koVabA"
        >
          July 9, 2021 7:17 PM
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          Testing
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          credit
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          AUD
        </td>
        <td
          class="sc-hKFxyN koVabA"
        >
          <label
            class="sc-fujyAs gyuDgd"
          >
            +$1,000
          </label>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
